js/components/portal-dashboard/feedback/activity-feedback-textarea.tsx

```diff
@@ -20,20 +21,29 @@ export const ActivityFeedbackTextarea: React.FC<IProps> = (props) => {
     }
   }, [textareaRef]);
 
-  const handleActivityFeedbackChange = (studentId: string | undefined) => (event: React.FormEvent<HTMLTextAreaElement>) => {
+  const handleActivityFeedbackChange = (event: React.FormEvent<HTMLTextAreaElement>) => {
+    const target = event.currentTarget as HTMLTextAreaElement;
+    setHeight(target.scrollHeight);
+    updateFeedbackThrottled();
+  };
+
+  const updateFeedback = () => {
     if (activityId && studentId && updateActivityFeedback) {
-      const target = event.currentTarget as HTMLTextAreaElement;
-      updateActivityFeedback(activityId, activityIndex, studentId, {feedback: target.value});
-      setHeight(target.scrollHeight);
+      updateActivityFeedback(activityId, activityIndex, studentId, {feedback: textareaRef.current?.value});
+      // eslint-disable-next-line no-console
+      console.log(textareaRef.current?.value);
     }
   };
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const updateFeedbackThrottled = useCallback(throttle(updateFeedback, 2000), []);
```

Comment:
I would think we would want debounce here rather than throttle. Throttle will save immediately on first change and then wait two seconds while debounce will wait until two seconds have elapsed before saving.

Reply:
I thought about using debounce at first too, but the story clearly states to save the user input data every 2 seconds.  With debounce, a user can continuously type for 60 seconds and none of their feedback is saved until after the typing is completed (so if they lost internet or their browser crashed at second 59, nothing is saved).  With throttle, we get the save every two seconds.  I also added an onBlur to deal with the final save if the user quickly moves off of the text area and navigates elsewhere.